src/lib/components/globalhealthbar/HistorySlider.tsx

```diff
@@ -0,0 +1,145 @@
+import styled, { css, useTheme } from 'styled-components';
+import { FormattedDateTime, Icon, spacing } from '../../index';
+import { FocusVisibleStyle } from '../buttonv2/Buttonv2.component';
+import { useHistoryAlert } from './HistoryProvider';
+import { getStep, setHistoryTooltipPosition } from './utils';
+import { useEffect } from 'react';
+
+const StyledRange = styled.input`
+  width: 600px;
+  padding: 0; /* nécessaire pour IE */
+  margin: 0;
+  margin-top: 2px;
+  appearance: none; /* nécessaire pour IE */
+  -moz-appearance: none; /* nécessaire pour Firefox */
+  -webkit-appearance: none; /* nécessaire pour Chrome */
+  font: inherit; /* même rendu suivant font document */
+  outline: none;
+  opacity: 1;
+  background: transparent; /* sert pour couleur de fond de la zone de déplacement */
+  box-sizing: content-box; /* même modèle de boîte pour tous */
+  transition: opacity 0.2s;
+  cursor: pointer;
+  position: absolute;
+  z-index: 10;
+  height: 16px;
+
+  /*==============================*/
+  /* cursor                       */
+  /*==============================*/
+  &::-webkit-slider-thumb {
+    -webkit-appearance: none;
+    padding: 0;
+    appearance: none;
+    margin: 0;
+    width: 2px;
+    height: 16px;
+    background-color: ${(props) => props.theme.selectedActive};
+  }
+  &::-moz-range-thumb {
+    margin: 0;
+    width: 2px;
+    height: 16px;
+    background-color: ${(props) => props.theme.selectedActive};
+    border: none;
+  }
+  :focus-visible::-webkit-slider-thumb {
+    ${FocusVisibleStyle}
+  }
+  :focus-visible::-moz-range-thumb {
+    ${FocusVisibleStyle}
+  }
+`;
+
+const HistoryContainer = styled.div`
+  width: 100%;
+  position: relative;
+`;
+
+const HistoryTooltipContainer = styled.div<{ inset }>`
+  position: absolute;
+  display: ${(props) => (props.inset ? 'flex' : 'none')};
+  inset: ${(props) => props.inset};
+  align-items: center;
+  flex-direction: column;
+  gap: ${spacing.r2};
+`;
+
+const HistoryTooltip = styled.div`
+  ${(props) => {
+    const theme = useTheme();
+    return css`
+      padding: ${spacing.r4} ${spacing.r8};
+      white-space: 'nowrap';
+      border: 1px solid ${theme.border};
+      border-radius: 4px;
+      width: 116px;
+      color: ${theme.textSecondary};
+    `;
+  }}
+`;
+type HistorySliderProps = {
+  start: string;
+  end: string;
+  startDate: number;
+  endDate: number;
+};
+
+export const HistoryAlertSlider = ({
```

Comment:
Similarly I think this could be used in other context than Alerting. It is a "Point in time Slider/Selector", not sure how to name it yet